pkg/controller/infinispan/infinispan_controller.go

```diff
@@ -1346,7 +1263,191 @@ func (r *ReconcileInfinispan) serviceExternal(m *infinispanv1.Infinispan) *corev
+	route := routev1.Route{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      ispn.GetServiceExternalName(),
+			Namespace: ispn.Namespace,
+			Labels:    lsService,
+		},
+		Spec: routev1.RouteSpec{
+			Host: ispn.Spec.Expose.Host,
+			To: routev1.RouteTargetReference{
+				Kind: "Service",
+				Name: ispn.Name},
+			TLS: &routev1.TLSConfig{Termination: routev1.TLSTerminationPassthrough},
+		},
+	}
+	return &route
+}
+
+// reconcileRoute creates the external Route if needed
+func (r *ReconcileInfinispan) reconcileRoute(ispn *infinispanv1.Infinispan, route *routev1.Route, logger logr.Logger) (*reconcile.Result, error) {
+	ro := &routev1.Route{}
+	ns := types.NamespacedName{
+		Name:      route.Name,
+		Namespace: route.Namespace,
+	}
+	err := r.client.Get(context.TODO(), ns, ro)
+	if errors.IsNotFound(err) {
+		controllerutil.SetControllerReference(ispn, route, r.scheme)
+		err = r.client.Create(context.TODO(), route)
+		if err != nil {
+			logger.Error(err, "failed to create Route", "Route", route)
+			return &reconcile.Result{Requeue: true, RequeueAfter: consts.DefaultRequeueOnCreateExposeServiceDelay}, nil
+		}
+		if route.Spec.Host != "" {
+			ispn.Spec.Expose.Host = route.Spec.Host
+			err = r.client.Update(context.TODO(), ispn)
+			if err != nil {
+				logger.Info("Failed to update Infinispan with route Host", "Route", route)
+			}
+		}
+	}
+	return nil, err
+}
+
+// computeIngress compute the Ingress object
+func (r *ReconcileInfinispan) computeIngress(ispn *infinispanv1.Infinispan, logger logr.Logger) *networkingv1beta1.Ingress {
+	lsService := ispncom.LabelsResource(ispn.ObjectMeta.Name, "infinispan-service-external")
+	ingress := networkingv1beta1.Ingress{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      ispn.GetServiceExternalName(),
+			Namespace: ispn.Namespace,
+			Labels:    lsService,
+		},
+		Spec: networkingv1beta1.IngressSpec{
+			TLS: []networkingv1beta1.IngressTLS{},
+			Rules: []networkingv1beta1.IngressRule{
+				{
+					Host: ispn.Spec.Expose.Host,
+					IngressRuleValue: networkingv1beta1.IngressRuleValue{
+						HTTP: &networkingv1beta1.HTTPIngressRuleValue{
+							Paths: []networkingv1beta1.HTTPIngressPath{
+								{
+									Path: "/",
+									Backend: networkingv1beta1.IngressBackend{
+										ServiceName: ispn.Name,
+										ServicePort: intstr.IntOrString{IntVal: consts.InfinispanPort}}}}},
+					}}},
+		}}
+	if ispn.Spec.Security.EndpointEncryption.CertSecretName != "" {
+		ingress.Spec.TLS = []networkingv1beta1.IngressTLS{
+			{
+				Hosts: []string{ispn.Spec.Expose.Host},
+			}}
+	}
+	return &ingress
+}
+
+// reconcileIngress creates the external Route if needed
+func (r *ReconcileInfinispan) reconcileIngress(ispn *infinispanv1.Infinispan, ingress *networkingv1beta1.Ingress, logger logr.Logger) (*reconcile.Result, error) {
+	i := &networkingv1beta1.Ingress{}
+	ns := types.NamespacedName{
+		Name:      i.Name,
+		Namespace: i.Namespace,
```

Comment:
```suggestion
		Namespace: ingress.Namespace,
```
But for me 'i' and 's' are not so informative

Reply:
the scope of these vars is very limited. I don't think they can generate confusion.

Reply:
Yes, sure but it was about search by name/namespace for `ingress` but not `i`

Reply:
ah yeah sure. thanks!